vault/core.go

```diff
@@ -2638,47 +2638,49 @@ func (c *Core) runUnsealSetupForPrimary(ctx context.Context, logger log.Logger)
 		return err
 	}
 
-	// Retrieve the seal generation information from storage
-	existingGenerationInfo, err := PhysicalSealGenInfo(ctx, c.physical)
-	if err != nil {
-		logger.Error("cannot read existing seal generation info from storage", "error", err)
-		return err
-	}
-
-	sealGenerationInfo := c.seal.GetAccess().GetSealGenerationInfo()
-
-	switch {
-	case existingGenerationInfo == nil:
-		// This is the first time we store seal generation information
-		fallthrough
-	case existingGenerationInfo.Generation < sealGenerationInfo.Generation:
-		// We have incremented the seal generation
-		if err := c.SetPhysicalSealGenInfo(ctx, sealGenerationInfo); err != nil {
-			logger.Error("failed to store seal generation info", "error", err)
+	if c.IsMultisealEnabled() {
+		// Retrieve the seal generation information from storage
+		existingGenerationInfo, err := PhysicalSealGenInfo(ctx, c.physical, c.IsMultisealEnabled())
```

Comment:
Mainly for my information, if we did have an existingGenerationInfo stored, I assume at this point we have validated that it contains only a single seal. 

As we don't seem to leverage this information or reset it, will this info be out of date, if the customer went from multiple seals -> single seal -> disable multiseal -> old style migration of seal migration -> re-enable multiseal?

Reply:
Yeah you won't have gotten as far as this due to the earlier check.   Fair point on the existing storage though.  May want to chat about this one.  We probably want to write a new seal gen info when multiseal is disabled successfully.

Reply:
Ugh, it's worse than that as you call out.  Almost as if when you enable multiseal you need to take the current configuration as gospel and save it, but that means you need to know it was off before.